source/developers-guide/example-plugin/index.md

```diff
@@ -28,187 +26,201 @@ This example gives answers to the following questions:
 * How to structure my plugin templates, so that they can be overwritten by local themes or other plugins?
 * How to override plugin templates inside my custom theme?
 
-## Bootstrap
+## Register template first
 
-```php
-
-<?php
+<div class="alert alert-info">
+<strong>Always register your template directory first to prevent smarty security exceptions</strong>
+<br>
+The plugin template directory should always be registered, even if it's not used in the current controller context. If your registration of the template depends on certain conditions, a Smarty Security error may occur. This could be confusing and annoying for other third-party developers, because it's not obvious where the error came from in the first place.
+</div>
 
-use ShopwarePlugins\SwagPluginSystem\StoreFrontBundle\ListProductService;
-use ShopwarePlugins\SwagPluginSystem\StoreFrontBundle\SeoCategoryService;
+First we register a new subscriber class for this.
 
-class Shopware_Plugins_Frontend_SwagPluginSystem_Bootstrap
-    extends Shopware_Components_Plugin_Bootstrap
-{
-    public function getLabel()
-    {
-        return 'Shopware 5 - Big picture of the plugin system';
-    }
+```xml
+<!-- Register TemplateRegistration subscriber -->
+<service id="swag_plugin_system.subscriber.template_registration" class="SwagPluginSystem\Subscriber\TemplateRegistration">
+    <argument>%swag_plugin_system.plugin_dir%</argument>
+    <argument type="service" id="template"/>
+    <tag name="shopware.event_subscriber"/>
+</service>
+```
+The Subscriber class requires 2 parameter in the constructor. 
+1. The Plugin base path like `../shopware/custom/plugins/SwagPluginSystem`.
+2. The Enlight_Template_Manager from the service container
 
-    public function install()
-    {
-        $this->subscribeEvent(
-            'Enlight_Bootstrap_InitResource_shopware_storefront.seo_category_service',
-            'registerSeoCategoryService'
-        );
+Last but not least we add the `shopware.event_subscriber` tag. 
+So that's clear that this class implements the Subscriber interface and shopware finds subscribed events and they handler.
```

Comment:
With this tag it is pointed out that the class implements the Subscriber interface and could be handled by Shopware to register new handler for certain events.

Reply:
d